### ISSUE TITLE
float is precision is not precise enough

### ISSUE THREAD

**Issue by rwols** [jbeder/yaml-cpp#469]
When I serialize the float `0.12345679` it gets serialized as `0.12345678`.

This fails: (include the header `boost/math/special_functions/next.hpp`)

```c++
float x = 0.12345679f;
float another_x;
{
    std::ofstream output("test.yml");
    YAML::Emitter emitter(output);
    emitter << YAML::BeginDoc 
            << YAML::BeginMap 
            << YAML::Key << "x" 
            << YAML::Value << x 
            << YAML::EndMap 
            << YAML::EndDoc;
}
{
    std::ifstream input("test.yml");
    const auto doc = YAML::Load(input);
    another_x = doc["x"].as<float>();
}
CHECK_TRUE(std::abs(boost::math::float_distance(x, another_x)) < 1);
```

**Comment by rwols** [comment]
I'd like to note that `std::abs(boost::math::float_distance(x, another_x)) < 3` DOES succeed

**Comment by jbeder** [comment]
Fixed by #649.
